dvclive/huggingface.py

```diff
@@ -0,0 +1,31 @@
+from transformers import (
+    TrainerCallback,
+    TrainerControl,
+    TrainerState,
+    TrainingArguments,
+)
+
+import dvclive
+
+
+class DvcLiveCallback(TrainerCallback):
+    def __init__(self, model_file=None):
+        super().__init__()
+        self.model_file = model_file
+
+    def on_evaluate(
```

Comment:
Btw, is there a reason why the Callbacks that are part of HuggingFace use `on_log` but we are using `on_evaluate`:

https://huggingface.co/transformers/_modules/transformers/integrations.html#CometCallback

This callback does look simpler than those other implementations, but I'm just curious about that

Reply:
Thank you for your comment!

PR in dvc.org: https://github.com/iterative/dvc.org/pull/2718

Yes, there is a reason why the Callbacks that are part of HuggingFace use `on_log` instead of `on_evaluate`. As you can see in https://huggingface.co/transformers/main_classes/callback.html#trainercallback, metrics are only available in the `on_evaluate` event. Thus, we can get the loss values both at the train and eval from `on_log`, but eval metrics only from `on_evaluate`.

The `on_evaluate` event has its disadvantage: we cannot get the training loss during this event, only the evaluation loss.

Reply:
Thanks for the explanation. Would it make sense to extend `DvcLiveCallback` to  use both `on_log` (for trainand eval loss) and `on_evaluate` (for eval metrics) ?

Reply:
Yes, I think it's a good idea. I did it

Reply:
I think that the huggingface docs might be a little misleading. I just debug the test locally and it seems that the same values available as `metrics` inside `on_evaluate` are previously being passed as `logs` to `on_log`.

Reply:
Yes, you're right, it's strange, but metrics logs are also written in `on_log` event. Thank you for your comment!

Reply:
Hmm, now for different operating systems, different `len(first(logs.values()))` values in the test are obtained

Reply:
Well that is strange, I will debug locally the macOS test case which is the one failing

Reply:
Still don't understand what the bug was. If I have to guess might be related with `first` behaving differently in MacOS. Anyway I made the test more explicit and it now passes 🎉